dora/underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -140,6 +146,21 @@ protected HdfsUnderFileSystem(AlluxioURI ufsUri, UnderFileSystemConfiguration co
       Configuration hdfsConf, boolean useLoadingCache) {
     super(ufsUri, conf);
 
+    mTrashEnable = alluxio.conf.Configuration.getBoolean(
+        PropertyKey.UNDERFS_HDFS_TRASH_ENABLE);
+    LOG.info(PropertyKey.UNDERFS_HDFS_TRASH_ENABLE.getName() + " is set to {}", mTrashEnable);
+    mFsTrash = CacheBuilder.newBuilder().build(new CacheLoader<FileSystem, Optional<Trash>>() {
+      @Override
+      public Optional<Trash> load(FileSystem fs) throws Exception {
+        Configuration configuration = fs.getConf();
+        long interval = configuration.getLong(CommonConfigurationKeys.FS_TRASH_INTERVAL_KEY, 0);
+        if (interval == 0) {
+          return Optional.empty();
+        }
+        return Optional.of(new Trash(fs, configuration));
+      }
+    });
```

Comment:
```suggestion
    mFsTrash = CacheBuilder.newBuilder().build(new CacheLoader<FileSystem, Optional<Trash>>() {
      @Override
      public Optional<Trash> load(FileSystem fs) throws Exception {
        return Optional.of(new Trash(fs, fs.getConf()));
      }
    });
```

Reply:
@dbw9580 How do you think?

Reply:
Looks good except that the `Optional` part can be removed since the value is always present.

Reply:
see this pr: https://github.com/huanghua78/alluxio/pull/1 @huanghua78 @dbw9580

Reply:
@humengyu2012 OK. I merged your modification manually into this PR.